### PR TITLE
auction(pcli): auction support in pcli

### PR DESCRIPTION
**Pull request by TalDerei**
## Describe your changes
Auction support in pcli

## Issue ticket number and link
References https://github.com/penumbra-zone/penumbra/issues/4242

## Checklist before requesting a review

- [x] If this code contains consensus-breaking changes, I have added the "consensus-breaking" label. Otherwise, I declare my belief that there are not consensus-breaking changes, for the following reason:

### CODE REVIEW

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [inline crates/bin/pcli/src/command/tx.rs:370]: 👍

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [inline crates/bin/pcli/src/command/tx.rs:236]:
```suggestion
        receive: String,
```

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [inline crates/bin/pcli/src/command/tx.rs:1424]: FWIW, we can do away with the boilerplate and make those branches `todo!()`, your call.

**Comment by TalDerei** [comment]
this will additionally encapsulate necessary changes to the pcli docs.

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [inline crates/bin/pcli/src/command/tx.rs:1345]: The goal of these commands is to be able to exercise the DA logic, IMO it would be clearer to have them take the exact fields that an auction description specifies i.e.

- a start height
- an end height
- a step count
- an input value
- a minimum output
- a maximum output

ditto for the rest of the commands. We are the customers for this command so being able to easily specify custom descriptions is a feature

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [inline crates/bin/pcli/src/command/tx.rs:1358]: Let's make this a flag `--end-height`

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [inline crates/bin/pcli/src/command/tx.rs:274]: minor point:
```suggestion
        /// Source address withdrawing from the auction.
```

**Comment by TalDerei** [comment]
flagging this workstream also has a dependency on having planner support https://github.com/penumbra-zone/penumbra/issues/4241

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [inline crates/bin/pcli/src/command/tx.rs:227]: Could we refactor the `DutchAuction*` variants into an `AuctionCmd` enum so that we can have these commands:

```rust
pcli auction dutch schedule ...
```

rather than:

`pcli dutch-auction-schedule`

🙏

**Review by erwanor — CHANGES_REQUESTED** [review]
Thank you for the work you have put in this, I added a few comments that I think will make testing the core mechanism easier for us. I think at this point we just need to get the basic interface right, so if you want to replace the command impls with `todo!()` that would be ok.

File: crates/bin/pcli/src/command/tx.rs
Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs:236]: fixed typo

File: crates/bin/pcli/src/command/tx.rs
Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs:1424]: what if we merge #4248 first?

File: crates/bin/pcli/src/command/tx.rs
Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs:1345]: ah good call, I was thinking of the customer in the context of the end user rather than us. It makes sense since we're still in dev mode. I'll make the necessary changes

File: crates/bin/pcli/src/command/tx.rs
Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs:274]: fixed

File: crates/bin/pcli/src/command/tx.rs
Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs:227]: I'll make the change. I originally did that, but then reverted since I noticed the issue description used `pcli tx auction`

File: crates/bin/pcli/src/command/tx.rs
Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs:1424]: I think the only boilerplate we should keep is the planner related code since we'll be passing all the necessary CLI parameters anyways.

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [reply crates/bin/pcli/src/command/tx.rs:227]: sorry I missed the tx in the command but we definitely want that

Comment (erwanor) [reply crates/bin/pcli/src/command/tx.rs:227]: the ask here is folding the auction commands into an AuctionCmd enum

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [reply crates/bin/pcli/src/command/tx.rs:1424]: Sounds good

File: crates/bin/pcli/src/command/tx.rs
Comment (erwanor) [reply crates/bin/pcli/src/command/tx.rs:227]: so just to clarify: pcli tx auction dutch ...

Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs:227]: yeah noted, was just clarifying!

File: crates/bin/pcli/src/command/tx.rs
Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs]: modified fields to match design doc

File: crates/bin/pcli/src/command/tx.rs
Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs]: @erwanor there seems to be something wrong with the clap ordering, can you check it out?

Comment (erwanor) [reply crates/bin/pcli/src/command/tx.rs]: @TalDerei i'm afk, but can take a look later, what's the problem exactly?

Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs]: when we run `pcli transaction auction dutch...`, the commands don't display the proper flags

Comment (TalDerei) [reply crates/bin/pcli/src/command/tx.rs]: I figured it out, using clap's `default_value = "0"` prevents the flag from being displayed in the CLI.

**Comment by erwanor** [comment]
LGTM, when we fill in the implementations we will want to ditch the nonce flag and use `OsRng + getrandom` to create and fill the buffer on behalf of the user. Let's do this when we get to it though!

**Review by erwanor — APPROVED** [review]

**Pull request merged**
